zstd.go

```diff
@@ -1,11 +1,21 @@
 package sarama
 
 import (
+	"runtime"
 	"sync"
+	"sync/atomic"
 
 	"github.com/klauspost/compress/zstd"
 )
 
+var zstdEncoderConcurrent int32
+
+// SetZstdEncoderConcurrent allow to manage counter of encoders in zstd writer
+func SetZstdEncoderConcurrent(concurrent int) {
```

Comment:
it would be cool, but it seems that it requires major changes in the compression mechanism.
`ZstdEncoderParams` is used in `zstdCompress`, which called in `compress` function. Here we have common compression parameters as codec and level of compression. Adding here one more parameter, which applies only to the Zstd, seemed wrong to me. Also, it seems that this parameter should be set during initialization and not change depending on the message or something else in runtime.

Reply:
if you have any good vision of how to do it, I'm ready to implement it :)